inc/functions/files.php

```diff
@@ -656,38 +662,28 @@ function rocket_clean_home( $lang = '' ) {
 	*/
 	do_action( 'before_rocket_clean_home', $root, $lang ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals
 
-	// Delete homepage.
-	$files = glob( $root . '/{index,index-*}.{html,html_gzip}', GLOB_BRACE | GLOB_NOSORT );
-	if ( $files ) {
-		foreach ( $files as $file ) { // no array map to use @.
-			rocket_direct_filesystem()->delete( $file );
-		}
-	}
-
-	// Delete homepage pagination.
-	$dirs = glob( $root . '*/' . $GLOBALS['wp_rewrite']->pagination_base, GLOB_NOSORT );
-	if ( $dirs ) {
-		foreach ( $dirs as $dir ) {
-			rocket_rrmdir( $dir );
+	foreach ( _rocket_get_cache_dirs( $parse_url['host'], $cache_path ) as $dir ) {
+		$domain_entry = $dir . $parse_url['path'];
+		if ( ! $filesystem->exists( $domain_entry ) ) {
+			continue;
 		}
-	}
-
-	// Remove the hidden empty file for mobile detection on NGINX with the Rocket NGINX configuration.
-	$nginx_mobile_detect_files = glob( $root . '/.mobile-active', GLOB_BRACE | GLOB_NOSORT );
-	if ( $nginx_mobile_detect_files ) {
-		foreach ( $nginx_mobile_detect_files as $nginx_mobile_detect_file ) { // no array map to use @.
-			rocket_direct_filesystem()->delete( $nginx_mobile_detect_file );
+		$iterator = new DirectoryIterator( $domain_entry );
```

Comment:
You should wrap this in a `try/catch` because the constructor can throw an exception: https://www.php.net/manual/en/directoryiterator.construct.php